### PR TITLE
Fix test failing caused by math agent and output parsers

### PR DESCRIPTION
**Pull request by ChenKua**
# Your checklist for this pull request
Thank you for submitting a pull request! To speed up the review process, please follow this checklist:

- [x] My Pull Request is small and focused on one topic so it can be reviewed easily 
- [x] My code follows the style guidelines of this project (`make format`)
- [x] Commit messages are detailed
- [x] I have performed a self-review of my code
- [x] I commented hard-to-understand parts of my code
- [x] I updated the documentation (docstrings, `/docs`)
- [x] My changes generate no new warnings (or explain any new warnings and why they're ok)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass when I run `pytest tests` (offline mode)

Additional steps for code with networking dependencies:

- [ ] I followed the [offline and online testing guidelines](https://sherpa-ai.readthedocs.io/en/latest/How_To/Test/test.html#offline-and-online-testing)
- [ ] All tests pass when I run `pytest tests --external_api` (online mode, making network calls)

# Description
Describe your pull request here.

What does this PR implement or fix? Explain.

If this PR resolves any currently open issues then mention them like this: `Closes #xxx`.
Github will close such issues automatically when your PR is merged into `main`.

Any relevant logs, error output, etc?

Any other comments? For example, will other contributors need to install new libraries via `poetry install` after picking up these changes?

💔Thank you!

### CODE REVIEW

File: src/sherpa_ai/agents/mathematician.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import os
```
Comment (20001LastOrder) [inline src/sherpa_ai/agents/mathematician.py:1]: In general, I think the mathematician is not too different from other agents except for the action of solving a math problem. I think we should implement that as an action so it can be configured into an existing agent to make that agent a _mathematician_.

Comment (ChenKua) [reply src/sherpa_ai/agents/mathematician.py:1]: i add a new action answer_arithmetic.py
Do I need to remove the math agent? @20001LastOrder

File: src/sherpa_ai/actions/answer_arithmetic.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        openai.api_key = os.getenv("OPENAI_API_KEY")
+        interface = pal.interface.ProgramInterface(
+            model=MODEL, get_answer_expr="solution()", verbose=PAL_verbose
+        )
+
+        interface.run_with_dict = types.MethodType(run_with_dict, interface)
+
+        self.interface = interface
+
+    def execute(self, question, placeholder=False) -> str:
```
Comment (20001LastOrder) [inline src/sherpa_ai/actions/answer_arithmetic.py:111]: Maybe make this placeholder parameter to be a parameter of the class rather than the function. I think we probably don't want the agent to decide on whether to set placeholder.

File: src/sherpa_ai/agents/mathematician.py (hunk 1 above)
Comment (20001LastOrder) [reply src/sherpa_ai/agents/mathematician.py:1]: Yes, I think the agent can be removed. And later we will figure out a configuration for the mathematician,

File: src/sherpa_ai/actions/answer_arithmetic.py (hunk 2 above)
Comment (ChenKua) [reply src/sherpa_ai/actions/answer_arithmetic.py]: fixed

File: src/sherpa_ai/agents/mathematician.py (hunk 1 above)
Comment (ChenKua) [reply src/sherpa_ai/agents/mathematician.py]: fixed

File: src/tests/data/test_context_search_test_context_search_succeeds.jsonl [hunk 3]
```diff
@@ -1 +0,0 @@
-{"input": [{"text": "Role Description: The programmer receives requirements about a program and write it\\nTask: We need to render a highly complex 3D image on the solar system. We can use any publicly avaliable\\n    resources to achieve this task.\\n\\nRelevant Documents:\\n\\n\\n\\nReview and analyze the provided documents with respect to the task. Craft a concise and short, unified summary that distills key information that is most relevant to the task, incorporating reference links within the summary.\\nOnly use the information given. Do not add any additional information. The summary should be less than 5 setences\\n", "agent": "human"}], "output": "The task is to render a highly complex 3D image of the solar system. The programmer is required to use publicly available resources to achieve this task. No specific documents are provided for review and analysis.", "llm_name": "gpt-3.5-turbo"}
```
Comment (20001LastOrder) [inline src/tests/data/test_context_search_test_context_search_succeeds.jsonl:1]: Why is this file removed?

Comment (ChenKua) [reply src/tests/data/test_context_search_test_context_search_succeeds.jsonl:1]: i don't remember deleting it. will add it back

**Review by 20001LastOrder — APPROVED** [review]
